DOCSP-45198: Distinct values (#111)
mongodb/docs-ruby: source/read/distinct.txt

.. _ruby-distinct:

==============================
Retrieve Distinct Field Values
==============================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference
 
.. meta::
   :keywords: read, unique, code example

Overview
--------

In this guide, you can learn how to use the {+driver-short+} to retrieve the
distinct values of a specified field across a collection.

> Within a collection, different documents might contain different values for a

Reviewer: ```suggestion
Within a collection, documents might contain different values for a
```

single field. For example, one document in a ``restaurants`` collection has a
``borough`` value of ``'Manhattan'``, and another has a ``borough`` value of

> ``'Queens'``. By using the {+driver-short+}, you can retrieve all the unique values
> that a field contains across multiple documents in a collection.

Reviewer: ```suggestion
``'Queens'``. You can use the {+driver-short+} to retrieve all the unique values
that a field contains across multiple documents in a collection.
```

Sample Data
~~~~~~~~~~~

The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
from your {+language+} application, create a ``Mongo::Client`` object that connects to
an Atlas cluster and assign the following values to your ``database`` and ``collection``
variables:

.. literalinclude:: /includes/read/distinct.rb

>     :language: ruby

Reviewer: S: fix indentation
